### PR TITLE
Add permutation_of_string.c

### PR DESCRIPTION
**Pull request by alvinuday**
Permutations of a given string in C

# Description

Please include a summary of the changes you have made in regard with this issue. 
C code for permutations of a given string

Fixes #(issue number)
#401 
## Type of change

**Please delete options that are not relevant**

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] This change requires a documentation update

# Checklist:

-   [x] I have read the contribution guidelines before making this PR.
-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [x] I have commented in my code, particularly in hard-to-understand areas
-   [x] I have made corresponding changes to the documentation (Language folder or any other readme instructed to)
-   [x] My changes generates no new warnings
-   [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by alvinuday** [comment]
is something wrong with the code? @kaustubhgupta

**Comment by alvinuday** [comment]
Screen Shot 2020-10-16 at 7 41 17 PM

I think the permutations are fine.
if there is any problem please revert

**Comment by alvinuday** [comment]
what improvement is needed? @kaustubhgupta @YashSaxena9

**Comment by kaustubhgupta** [comment]
@alvinuday @YashSaxena9 Will suggest changes. Wait for him to review your PR

File: C/permutation_of_string.c [hunk 1]
```diff
@@ -0,0 +1,35 @@
+#include <stdio.h>
```
Comment (kaustubhgupta) [inline C/permutation_of_string.c:1]: update C folder readme file

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

**Comment by kaustubhgupta** [comment]
The PR is not linked to any issue, kindly fix that

**Pull request closed without merging**